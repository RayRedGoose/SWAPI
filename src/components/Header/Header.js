import React from 'react'
import './Header.scss'
import UserProfile from '../UserProfile/UserProfile'
import BackButton from '../BackButton/BackButton'

const Header = (props) => {
  const { headerText, user, logOut, isButton } = props
  return (
    <header className="header">
      {isButton && <BackButton />}
      <h2>{headerText}</h2>
      <UserProfile
        {...user}
        logOut={logOut}
      />
    </header>
  )
}

export default Header
