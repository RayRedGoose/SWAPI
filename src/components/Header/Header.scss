.header {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  padding: 40px;
  h2 {
    font-size: 3.2em;
  }
}
