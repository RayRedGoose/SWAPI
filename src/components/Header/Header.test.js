import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'

describe('Header', () => {

  it('should render Header with all data passed in correctly', () => {
    const wrapper = shallow(
      <Header
        user={{
          name: 'Taylor Swift',
          quote: "I find your lack of faith disturbing",
          rank: 'intermediate'}}
        logOut={jest.fn()}
        headerText={'Episode 1: Revenge of the Sloths'}
        isButton={true}
      />)

    expect(wrapper).toMatchSnapshot()
  })
})
