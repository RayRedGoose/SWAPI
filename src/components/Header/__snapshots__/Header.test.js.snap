// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header with all data passed in correctly 1`] = `
<header
  className="header"
>
  <BackButton />
  <h2>
    Episode 1: Revenge of the Sloths
  </h2>
  <UserProfile
    logOut={[MockFunction]}
    name="Taylor Swift"
    quote="I find your lack of faith disturbing"
    rank="intermediate"
  />
</header>
`;
